src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionTelemetryManager.cs

```diff
@@ -0,0 +1,107 @@
+        {
+            this.requestData = requestData;
+        }
+
+        /// <inheritdoc/>
+        public void OnEnvironmentVariableAdded(DataCollectorInformation dataCollectorInformation, string name, string value)
+        {
+            AddProfilerMetricForNewVariable(CorProfilerVariable, CorProfilerTelemetryTemplate, dataCollectorInformation, name, value);
+            AddProfilerMetricForNewVariable(CoreClrProfilerVariable, CoreClrProfilerTelemetryTemplate, dataCollectorInformation, name, value);
+        }
+
+        /// <inheritdoc/>
+        public void OnEnvironmentVariableConflict(DataCollectorInformation dataCollectorInformation, string name, string existingValue)
+        {
+            AddProfilerMetricForConflictedVariable(CorProfilerVariable, CorProfilerTelemetryTemplate, dataCollectorInformation, name, existingValue);
+            AddProfilerMetricForConflictedVariable(CoreClrProfilerVariable, CoreClrProfilerTelemetryTemplate, dataCollectorInformation, name, existingValue);
+        }
+
+        /// <inheritdoc/>
+        public IRequestData RequestData => requestData;
+
+        private void AddProfilerMetricForNewVariable(string profilerVariable, string telemetryTemplateName, DataCollectorInformation dataCollectorInformation, string name, string value)
+        {
+            if (!string.Equals(profilerVariable, name, StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            requestData.MetricsCollection.Add(string.Format(telemetryTemplateName, dataCollectorInformation.DataCollectorConfig?.TypeUri?.ToString()), GetProfilerName(value));
+        }
+
+        private void AddProfilerMetricForConflictedVariable(string profilerVariable, string telemetryTemplateName, DataCollectorInformation dataCollectorInformation, string name, string existingValue)
+        {
+            if (!string.Equals(profilerVariable, name, StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            if (string.Equals(existingValue, ClrIeProfilerGuid, StringComparison.OrdinalIgnoreCase))
+            {
+                if (dataCollectorInformation.TestExecutionEnvironmentVariables != null &&
+                    dataCollectorInformation.TestExecutionEnvironmentVariables.Any(pair => pair.Key.StartsWith(ClrIeInstrumentationMethodConfigurationPrefix32Variable)) &&
+                    dataCollectorInformation.TestExecutionEnvironmentVariables.Any(pair => pair.Key.StartsWith(ClrIeInstrumentationMethodConfigurationPrefix64Variable)))
+                {
+                    requestData.MetricsCollection.Add(string.Format(telemetryTemplateName, dataCollectorInformation.DataCollectorConfig?.TypeUri?.ToString()), ClrIeProfilerName);
+                    return;
+                }
+            }
+
+            requestData.MetricsCollection.Add(string.Format(telemetryTemplateName, dataCollectorInformation.DataCollectorConfig?.TypeUri?.ToString()), OverwrittenProfilerName);
```

Comment:
you mean profiler that we overwrite?

Reply:
Added information which profiler was overwritten. So for example we can get overwritten(vanguard). It means that dc wanted vanguard but other data collector already used other profiler.

Reply:
COR_PROFILER: If the COR_ENABLE_PROFILING check passes, the CLR connects to the profiler that has this CLSID or ProgID, which must have been stored previously in the registry. The COR_PROFILER environment variable is defined as a string, as shown in the following two examples.

C++

set COR_PROFILER={32E2F4DA-1BEA-47ea-88F9-C5DAF691C94A}  
set COR_PROFILER="MyProfiler"

Based on this it can be some name. I think we should not collect this.

Reply:
I could add detection if this is in form of {GUID} and in such case always log it. So we could remove whole logic of detecting clrie, intellitrace or covrun. WDYT?

It would be for example:
key=VS.TestPlatform.DataCollector.CoreClrProfiler.datacollector://microsoft/CodeCoverage/2.0
value={32E2F4DA-1BEA-47ea-88F9-C5DAF691C94A}

Reply:
I've changed the logic to use guids. When overwriting added both guids. If can't parse to guid we use empty guid.

Reply:
That does seem better, yes.